drivers/accel/adxl313/iio_adxl313.c

```diff
@@ -0,0 +1,1200 @@
+		counter ++;
+	if (mask & NO_OS_BIT(1))
+		counter ++;
+	if (mask & NO_OS_BIT(2))
+		counter ++;
+	if (mask & NO_OS_BIT(3))
+		counter ++;
+
+	iio_adxl313->no_of_active_channels = counter;
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Handles trigger: reads one data-set and writes it to the buffer.
+ *
+ * @param dev_data  - The iio device data structure.
+ *
+ * @return ret - Result of the handling procedure. In case of success, the size
+ * 				 of the written data is returned.
+*******************************************************************************/
+static int32_t adxl313_trigger_handler(struct iio_device_data *dev_data)
+{
+	int32_t data_buff[3];
+	int16_t x,y,z;
+	uint8_t i = 0;
+
+	struct adxl313_iio_dev *iio_adxl313;
+	struct adxl313_dev *adxl313;
+
+	if (!dev_data)
+		return -EINVAL;
+
+	iio_adxl313 = (struct adxl313_iio_dev *)dev_data->dev;
+
+	if (!iio_adxl313->adxl313_dev)
+		return -EINVAL;
+
+	adxl313 = iio_adxl313->adxl313_dev;
+
+	adxl313_get_raw_xyz(adxl313, &x, &y, &z);
+
+	if (dev_data->buffer->active_mask & NO_OS_BIT(0)) {
+		data_buff[0] = no_os_sign_extend32(x, 12);
+		i++;
+	}
+	if (dev_data->buffer->active_mask & NO_OS_BIT(1)) {
+		data_buff[i] = no_os_sign_extend32(y, 12);
+		i++;
+	}
+	if (dev_data->buffer->active_mask & NO_OS_BIT(2)) {
+		data_buff[i] = no_os_sign_extend32(z, 12);
+		i++;
+	}
+
+	return iio_buffer_push_scan(dev_data->buffer, &data_buff[0]);
+}
+
+/***************************************************************************//**
+ * @brief Initializes the ADXL313 IIO driver
+ *
+ * @param iio_dev    - The iio device structure.
+ * @param init_param - The structure that contains the device initial
+ * 		       		   parameters.
+ *
+ * @return ret       - Result of the initialization procedure.
+*******************************************************************************/
+int adxl313_iio_init(struct adxl313_iio_dev **iio_dev,
+		     struct adxl313_iio_dev_init_param *init_param)
+{
+	int ret;
+	struct adxl313_iio_dev *desc;
+
+	desc = (struct adxl313_iio_dev *)calloc(1, sizeof(*desc));
+	if (!desc)
+		return -ENOMEM;
+
+	desc->iio_dev = &adxl313_iio_dev;
+
+	// Initialize ADXL314 driver
+	ret = adxl313_init(&desc->adxl313_dev, *(init_param->adxl313_dev_init));
+	if (ret)
+		goto error_adxl313_init;
+
+	/* Set the range to +/- 12 g for ADXL312 and +/- 4 g for ADXL313 (maximum range) */
+	if (desc->adxl313_dev->dev_type == ID_ADXL312) {
+		ret = adxl313_set_range(desc->adxl313_dev, ADXL313_12G_RANGE);
+		if (ret)
+			goto error_config;
+	} else {
+		if (desc->adxl313_dev->dev_type == ID_ADXL313) {
+			ret = adxl313_set_range(desc->adxl313_dev, ADXL313_4G_RANGE);
+			if (ret)
+				goto error_config;
+		}
+	}
+
+	/* Enables full resolution for ADXL312 and ADXL313 */
+	if (desc->adxl313_dev->dev_type != ID_ADXL314) {
+		ret = adxl313_enable_full_res(desc->adxl313_dev);
+		if (ret)
+			return ret;
+	}
+
+	// Set operation mode
```

Comment:
nitpick: use the same comment format for line comments. check through the file.
suggestion: use /* */ for line comments (format used for Linux development)

Reply:
Thank you! I went through all the file and corrected the format.